### PR TITLE
Heartbeats chain simulator

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Enabled `/node/heartbeatstatus` endpoint in chain simulator

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sstanculeanu — APPROVED** [review]

File: node/chainSimulator/components/nodeFacade.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package components
 import (
 	"errors"
 	"fmt"
+	"github.com/multiversx/mx-chain-go/factory"
+	heartbeat2 "github.com/multiversx/mx-chain-go/node/chainSimulator/components/heartbeat"
```
Comment (sstanculeanu) [inline node/chainSimulator/components/nodeFacade.go:7]:
```suggestion
	simulatorHeartbeat "github.com/multiversx/mx-chain-go/node/chainSimulator/components/heartbeat"
```

Comment (miiu96) [reply node/chainSimulator/components/nodeFacade.go]: done

**Comment by miiu96** [comment]
Run Tests:
mx-chain-simulator-go: update-barnard
mx-chain-testing-suite: rc/barnard

File: node/chainSimulator/process/processor.go [hunk 2]
```diff
@@ -141,6 +151,32 @@ func (creator *blocksCreator) CreateNewBlock() error {
 	return creator.nodeHandler.GetBroadcastMessenger().BroadcastTransactions(transactions, blsKey.PubKey())
 }
 
+func (creator *blocksCreator) setHeartBeat(header data.HeaderHandler) error {
```
Comment (ssd04) [inline node/chainSimulator/process/processor.go:154]: there is no err return on this function, so it can be without a return value

**Review by ssd04 — APPROVED** [review]

**Review by elvisboghiu — APPROVED** [review]

**Pull request merged**